server/market/market.go

```diff
@@ -1164,19 +1164,19 @@ func (m *Market) collectPreimages(orders []order.Order) (cSum []byte, ordersReve
 		// The clients preimage response comes back via a channel, where nil
 		// indicates client failure to respond, either due to disconnection or
 		// no action.
-		piChan := make(chan *order.Preimage)
+		piChan := make(chan *order.Preimage, 1) // buffer so the link's in handler does not block
 
 		reqData := &piData{
 			ord:      ord,
 			preimage: piChan,
 		}
 
 		// Failure to respond in time is a miss, signalled by a nil pointer.
-		var missOnce sync.Once
 		miss := func() {
 			// RequestWithTimeout should only call this on expire, not because
 			// of link or other errors, but put it in a sync.Once to be safe
 			// since it will block if run twice.
+			var missOnce sync.Once
```

Comment:
Wasn't the purpose of the `sync.Once` to prevent sending `nil` on the `piChan` twice? It seems like you can do that after this change, since you're creating a new `sync.Once` every time `miss` is called.

Reply:
Yeah, this was a derp.  It totally needs to capture it.  Good catch.